### PR TITLE
docs(halo): Document device join flow (receiving)

### LINKED ISSUE #6482: [docs] document device join flow and API in typescript guide
Right now we're missing documentation and guide about how to invite and receive invites to device join flows in the TypeScript
```[tasklist]
### Tasks
- [x] Document receiving a device-join invitation from typescript
- [ ] Document creating a device-join invitation from typescript
```

### PR DESCRIPTION
**Pull request by ZaymonFC**
- Partially resolves #6482

### CODE REVIEW

File: docs/content/guide/halo/typescript.md [hunk 1]
```diff
@@ -57,3 +54,53 @@ const client = new Client();
...
 
 Before data can be manipulated, a [`space`](../echo/typescript/README.md) must be obtained.
+
+## Receiving a device join invitation
+We often want to add a device to the same [HALO identity](../halo/).
+You can accept a device join invitation by providing the invitation code and the auth code to the dxos client.
+This example demonstrates how to accept a device join invitation in a console application.
+A device join invitation can be obtained from [the shell](../halo/#shell) or generated in typescript from an existing identity.
+
+```ts{7}
```
Comment (zhenyasav) [inline docs/content/guide/halo/typescript.md:64]: make these snippets?

Comment (ZaymonFC) [reply docs/content/guide/halo/typescript.md:64]: Can do, although they don't seem to load and we duplicate the code across both the markdown and the snippet in the examples I've seen? I can look into it.

Comment (zhenyasav) [reply docs/content/guide/halo/typescript.md:64]: see any of the quickstarts for the code-fence syntax and then run `pnpm nx apidoc docs` to propagate the snippets into the md files

Comment (ZaymonFC) [reply docs/content/guide/halo/typescript.md:64]: Ahh. `apidoc docs` is the secret sauce I was missing.

File: docs/content/guide/halo/typescript.md [hunk 2]
```diff
@@ -57,3 +54,53 @@ const client = new Client();
...
+};
+
+/** Accepts a device join invitation by prompting the user for:
+ *    - invitation code
+ *    - authentication code */
+const acceptInvitation = async (client: Client) => {
+  const encodedInvitation = readlineSync.question('Please enter an invitation: ');
+  const deviceInvitationCode = getDeviceInvitationCode(encodedInvitation);
+
+  if (!deviceInvitationCode) {
```
Comment (zhenyasav) [inline docs/content/guide/halo/typescript.md:86]: we can try to simplify this snippet to be less "real" and more to the point ... e.g.: how about assuming the two values are coming from process.env somehow and are already parsed? parsing the URL might be a separate snippet in a collapsed detail section of the doc

Comment (ZaymonFC) [reply docs/content/guide/halo/typescript.md:86]: If you're accepting an invite from composer for example, you have to join the invitation before the shell UI presents you with the auth code.

File: docs/content/guide/halo/typescript.md [hunk 3]
```diff
@@ -57,3 +55,58 @@ const client = new Client();
...
+  } else {
+    const authCode = '123456'; // Take this as input from the user.
+    await invitationResult.authenticate(authCode);
+  }
+};
+```
+
+::: tip
+
+* Your [client](../../api/@dxos/client) instance needs to be [configured](../echo/typescript/config.md) correctly to talk to a [signalling service](../glossary.md#signaling-service).
```
Comment (zhenyasav) [inline docs/content/guide/halo/typescript.md:91]:
```suggestion
* Your [client](../../api/@dxos/client) instance needs to be [configured](../echo/typescript/config.md) correctly to talk to a [signaling service](../glossary.md#signaling-service).
```

**Review by zhenyasav — APPROVED** [review]

**Pull request merged**